addons/hr_contract/models/hr_employee.py

```diff
@@ -203,10 +203,19 @@ def _get_unusual_days(self, date_from, date_to=None):
         ])
         if not employee_contracts:
             return super()._get_unusual_days(date_from, date_to)
+
+        # Filter contracts that are "running" during this period
+        running_contracts = employee_contracts.filtered(lambda c: c.state == 'open')
+
+        if running_contracts:
+            selected_contract = running_contracts[0]
+        else:
+            # No "open" contract, pick the latest created contract from the valid ones
+            selected_contract = sorted(employee_contracts, key=lambda c: (c.create_date, c.id), reverse=True)[0]
```

Comment:
As employee can have at most one open contract at a time we do not need the `[0]`. I think this could be more efficient to use `max` instead of `sorted()[0]`
```suggestion
        selected_contract = employee_contracts.filtered(lambda c: c.state == 'open')

        if not selected_contract:
            selected_contract = max(employee_contracts, key=lambda c: (c.create_date, c.id))
```